flw/src/main/java/org/motechproject/nms/flw/service/impl/FrontLineWorkerServiceImpl.java

```diff
@@ -150,6 +163,69 @@ public State getState(FrontLineWorker frontLineWorker) {
         return state;
     }
 
+    @Override // NO CHECKSTYLE Cyclomatic Complexity
+    public boolean createUpdate(Map<String, Object> flw) {
+
+        long stateId = (long) flw.get(FlwConstants.STATE_ID);
+        long districtId = (long) flw.get(FlwConstants.DISTRICT_ID);
+        String mctsFlwId = flw.get(FlwConstants.ID).toString();
+        long contactNumber = (long) flw.get(FlwConstants.CONTACT_NO);
+
+        State state = locationService.getState(stateId);
+        if (state == null) {
+            flwErrorDataService.create(new FlwError(mctsFlwId, stateId, districtId, FlwErrorReason.INVALID_LOCATION_STATE));
+            return false;
+        }
+        District district = locationService.getDistrict(stateId, districtId);
+        if (district == null) {
+            flwErrorDataService.create(new FlwError(mctsFlwId, stateId, districtId, FlwErrorReason.INVALID_LOCATION_DISTRICT));
+            return false;
+        }
+
+        FrontLineWorker existingFlwByNumber = getByContactNumber(contactNumber);
+        FrontLineWorker existingFlwByMctsFlwId = getByMctsFlwIdAndState(mctsFlwId, state);
+        Map<String, Object> location = new HashMap<>();
+        try {
+            location = locationService.getLocations(flw);
+
+            if (existingFlwByMctsFlwId != null && existingFlwByNumber != null) {
+
+                if (existingFlwByMctsFlwId.getMctsFlwId().equalsIgnoreCase(existingFlwByNumber.getMctsFlwId()) &&
```

Comment:
This if check feels weird to me.  Basically there are two lookups to the database, one by mctsFlwId and the other by phone.  We then compare the results to verify the ids match and the two records are in the same state.  Is it possible to get two records back or can we just compare the primary key id?